File: warrior/src/main/kotlin/com/dumbdogdiner/warrior/kits/WarriorKit.java
package com.dumbdogdiner.warrior.kits;

import com.dumbdogdiner.warrior.api.kit.Kit;
import com.dumbdogdiner.warrior.api.kit.WithAbility;
import com.dumbdogdiner.warrior.api.user.WarriorUser;
import org.bukkit.Material;
import org.bukkit.entity.Player;
import org.bukkit.inventory.ItemStack;
import org.bukkit.potion.PotionEffect;
import org.bukkit.potion.PotionEffectType;

public class WarriorKit implements Kit, WithAbility {

    @Override
    public String getName() {
        return "Warrior";
    }

    @Override
    public String[] getDescription() {
        return new String[] {
            "&8Default Kit",
            " ",
            "&7&oBattle your way through the battle field",
            "&7&owith this trusty old sword. Perfect for",
            "&7&oa newcomer like you!",
            " ",
            "&7Main Weapon: &fIron Sword"
        };
    }

    @Override
    public void setupInventory(WarriorUser<?> user) {
        Player p = user.getBukkitPlayer();
        p.getInventory().clear();

        p.getInventory().setHelmet(new ItemStack(Material.LEATHER_HELMET));
        p.getInventory().setChestplate(new ItemStack(Material.CHAINMAIL_CHESTPLATE));
        p.getInventory().setLeggings(new ItemStack(Material.LEATHER_LEGGINGS));
        p.getInventory().setBoots(new ItemStack(Material.LEATHER_BOOTS));

        p.getInventory().setItem(0, new ItemStack(Material.IRON_SWORD));

        // TODO: How to handle Special Items???

        /**
         *         Material m = Material.FIREWORK_STAR;
         *         String decorator = Ability.DEACTIVATED_ABILITY_STRING;
         *         if(getAbility().availableOnStart()) {
         *             m = Material.MAGMA_CREAM;
         *             decorator = Ability.ACTIVE_ABILITY_STRING;
         *         }
         *
         *         ItemStack special = new ItemBuilder(m)
         *                             .setName(decorator)
         *                             .setLore("&7Activate your Special Ability")
         *                             .build();
         *
         *         p.getInventory().setItem(8, special);
         *         giveHealItems(p, 7);
         */
    }

    @Override
    public void execute(WarriorUser<?> user) {
        user.executeAsync((u) -> {
            Player player = u.getBukkitPlayer();

            for(PotionEffect potion : player.getActivePotionEffects()) {
                player.removePotionEffect(potion.getType());
            }

            player.addPotionEffect(new PotionEffect(PotionEffectType.REGENERATION, 100, 1));

        });
    }

    @Override
    public int getCooldown() {
        return 300;
    }

    @Override
    public boolean abilityReady(WarriorUser<?> user) {
        return !user.getAbilityActive();
    }
}

File: warrior/src/main/kotlin/com/dumbdogdiner/warrior/kits/WarriorKit.kt
package com.dumbdogdiner.warrior.kits

import com.dumbdogdiner.warrior.api.kit.Kit
import com.dumbdogdiner.warrior.api.kit.SlotEnum
import com.dumbdogdiner.warrior.api.kit.WarriorAbility
import com.dumbdogdiner.warrior.api.kit.weapons.ItemRarity
import com.dumbdogdiner.warrior.api.kit.weapons.WarriorWeapon
import com.dumbdogdiner.warrior.kits.abilities.MedicAbility
import org.bukkit.Material
import org.bukkit.inventory.ItemStack

class WarriorKit : Kit {

    private val primaryWeapon = WarriorWeapon(Material.IRON_SWORD)
                                .itemName("&8··· &7THE SWORD &8···")
                                // :eyes: .sweepAttack(SweepAttack.SHARP_STING)
                                .description("&7Rarity ${ItemRarity.COMMON.formatted}", " ", "&7&oDescription goes here...")
                                .damageOverride(6.0)
                                .hideFlags()
                                .build()

    override fun getName(): String {
        return "Warrior"
    }

    override fun getDescription(): Array<String> {
        return arrayOf(
                "Lorem ipsum dolor sit amet, consectetur ",
                "adipiscing elit. Pellentesque eu."
        )
    }

    override fun getIcon(): Material {
        return Material.IRON_SWORD
    }

    override fun getAbility(): WarriorAbility {
        return MedicAbility()
    }

    override fun getPrimary(): ItemStack {
        return this.primaryWeapon
    }

    override fun getItems(): Map<SlotEnum, ItemStack> {
        return mapOf(
                SlotEnum.PRIMARY to this.primary,
                SlotEnum.HELMET to ItemStack(Material.IRON_HELMET),
                SlotEnum.CHEST to ItemStack(Material.CHAINMAIL_CHESTPLATE),
                SlotEnum.LEGGINGS to ItemStack(Material.LEATHER_LEGGINGS),
                SlotEnum.BOOTS to ItemStack(Material.LEATHER_BOOTS)
        )
    }
}